pkg/controller/metrics/metrics.go

```diff
@@ -109,6 +109,14 @@ var (
 		},
 		[]string{"cluster_deployment", "namespace", "cluster_type"},
 	)
+
+	// MetricClusterPoolStaleClusterDeploymentsDeleted tracks the total number of CDs we delete
+	// because they became "stale". That is, the ClusterPool was modified in a substantive way such
+	// that these CDs no longer match its spec.
+	MetricClusterPoolStaleClusterDeploymentsDeleted = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "hive_clusterpool_stale_clusterdeployments_deleted",
+		Help: "The number of ClusterDeployments deleted because they no longer match the spec of their ClusterPool.",
+	}, []string{"clusterpool_namespace", "clusterpool_name"})
```

Comment:
As far as I know, it's ok and probably best to define and register these where they're used. This routine handles a lot of the gauges for which we can't assume a controller is reconciling reliably, so we run this block every 2m, but other controllers have metrics and they're defined in that package. 

If you're going to do a gauge, I think you might want to do it in this module though, as your reconcile loop is only for a specific pool, whereas we need to see the number across all pools.

Reply:
Got it.

I'll leave the gauge out for now. YAGNI. We can add it later if there's a call for it. (I'm kind of guessing this counter won't be used much either.)